pkg/utils/map.go

```diff
@@ -19,6 +19,8 @@ func MergeMap(sources ...map[string]string) map[string]string {
 // MergeMapNested will take two or more maps, merging the entries of the sources map into a
 // destination map. If both maps share the same key then destination's value for that key will be
 // merged with what's in source. The source maps are not modified.
+// If allowOverwrite is false, an error is returned if the key already exists and its value is not
+// nil.
 func MergeMapNested(allowOverwrite bool, sources ...map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
Now that our config is unstructured, do you think we could use this to merge cluster level and dc level Cassandra configs? Some folks have had issues with this as they were defining some settings at cluster level and others at dc level, but then all the cluster level ones were deleted (we're not doing any merge).

Reply:
That's a very good idea, so I created an issue: https://github.com/k8ssandra/k8ssandra-operator/issues/715

We must keep in mind though: JVM options are still structured, so we can't use this function for them.